packages/caravan-psbt/src/psbtv0/psbt.ts

```diff
@@ -265,3 +268,133 @@ const getHashForSignature = (
   }
   throw new Error("No redeem or witness script found for input.");
 };
+
+/**
+ * Given a string, try to create a Psbt object based on MAGIC (hex or Base64)
+ */
+export function autoLoadPSBT(psbtFromFile, options?: any) {
+  if (typeof psbtFromFile !== "string") return null;
+  // Auto-detect and decode Base64 and Hex.
+  if (psbtFromFile.substring(0, 10) === PSBT_MAGIC_HEX) {
+    return Psbt.fromHex(psbtFromFile, options);
```

Comment:
This is being done different than psbtv2 and checking only stringified magic bytes isn't good enough. The entire string must be checked for validity of its encoding and then check magic bytes as bytes. Why not try to unify? Below is sort-of how psbtv2 is doing it.

```
const psbtBuf = bufferize(psbtFromFile); // Validates hex or base64 and bufferizes string
const br = new BufferReader(buf);
if (!br.readBytes(PSBT_MAGIC_BYTES.length, true).equals(PSBT_MAGIC_BYTES)) {
  throw Error("Psbt magic bytes are incorrect.");
}

...

```

validateHex and validateBase64 are two functions which will tell you what you have. I like checking for encoding and then checking for magic bytes separately.